### PR TITLE
Bug 1204827 - Generate xClientState in SyncEngine, r=ferjm

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cf92e24e72a0e4258f64ec8a3f431bc3f68e4b13

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=899a2f4f557b8296798f42d95d9bdf3c8afee310

File: apps/sync/js/sync-engine/syncengine.js [hunk 1]
```diff
@@ -124,12 +124,28 @@ e!`);
     }
   });
 
+  const generateXClientState = (kB) => {
+    var kBarray = [];
+    for(var i = 0; i < kB.length; i+=2){
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: nit: spaces in `i += 2`

File: apps/sync/js/sync-engine/syncengine.js [hunk 2]
```diff
@@ -124,12 +124,28 @@ e!`);
...
+      kBarray.push(parseInt(kB.substring(i, i+2), 16));
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: same nit `i + 2`

File: apps/sync/js/sync-engine/syncengine.js [hunk 3]
```diff
@@ -124,12 +124,28 @@ e!`);
...
+    }
+    return window.crypto.subtle.digest({name: 'SHA-256'},
+                                       new Uint8Array(kBarray))
+    .then(hash => {
+      var bytes = new Uint8Array(hash).slice(0, 16);
+      var array = [];
+      bytes.forEach(function (byte){
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: `byte => {}`

File: apps/sync/js/sync-engine/syncengine.js [hunk 4]
```diff
@@ -275,7 +287,15 @@ uld be a Function`);
       if (this._ready) {
         return Promise.resolve();
       }
-      return this._syncCollection('meta').then(() => {
+      return generateXClientState(this._kB).then(xClientState => {
+        this._kinto = this._kinto = this._createKinto({
```
Comment (ferjm) [inline apps/sync/js/sync-engine/syncengine.js]: One `this._kinto` should make the work :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0122ab64776829108add54ed1469d01aae2d9cdb

**Comment by michielbdejong** [comment]
@ferjm addressed your review comments and rebased.

**Pull request merged**